.github/workflows/publish.yml

```
@@ -1,46 +1,74 @@
-name: Publish the how-to
+name: Publish Documentation
 
 on:
-  push:
-    branches:
-      - master
+  workflow_run:
+    workflows: ["Build Project"]
+    branches: [master]
+    types: [completed]
   schedule:
-    - cron:  '0 0 * * *'
+    - cron: '30 0 * * *'  # Run 30 minutes after the build workflow
+
+env:
+  JAVA_VERSION: '17'
+  MAVEN_VERSION: '3.9.5'
+  MAVEN_OPTS: '-Xmx3072m'
 
 permissions:
   contents: read
   pages: write
   id-token: write
+  actions: read
 
 jobs:
-  build-and-deploy:
+  publish:
+    name: Generate and publish documentation
     runs-on: ubuntu-latest
+    if: ${{ github.event.workflow_run.conclusion == 'success' || github.event_name == 'schedule' }}
+
     steps:
-      - uses: actions/checkout@v4
-      - uses: actions/setup-java@v4
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Download build artifacts
+        if: ${{ github.event_name != 'schedule' }}
+        uses: actions/download-artifact@v4
+        with:
+          name: build-artifacts-ubuntu-latest
+          path: target/
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          run-id: ${{ github.event.workflow_run.id }}
+
+      - name: Set up JDK ${{ env.JAVA_VERSION }}
+        if: ${{ github.event_name == 'schedule' }}
+        uses: actions/setup-java@v4
         with:
           distribution: 'temurin'
-          java-version: '17'
-      - name: Run builds
+          java-version: ${{ env.JAVA_VERSION }}
+
+      - name: Download artifacts fallback
+        if: ${{ github.event_name == 'schedule' }}
         run: |
-          mkdir --parents "$HOME/.local/bin"
-          curl -sL https://github.com/protocolbuffers/protobuf-javascript/releases/download/v3.21.4/protobuf-javascript-3.21.4-linux-x86_64.tar.gz | tar xzfO - bin/protoc-gen-js > "$HOME/.local/bin/protoc-gen-js"
-          chmod u+x "$HOME/.local/bin/protoc-gen-js"
-          curl -sL https://github.com/grpc/grpc-web/releases/download/1.5.0/protoc-gen-grpc-web-1.5.0-linux-x86_64 > "$HOME/.local/bin/protoc-gen-grpc-web"
-          chmod u+x "$HOME/.local/bin/protoc-gen-grpc-web"
-          ./.run.builds.sh
+          mvn -B clean install
```

```suggestion
          ./.run.builds.sh
```

If we keep the file we can call it. The upside of having the same structure everywhere is it's easier to upgrade all the how-tos at the same time when needed